px4_sitl_rtps doesn't work with px4_ros_com micrortps_bridge

## Describe the bug
In the `master` branch of this repo (from about a month ago), I could follow the steps described [here](https://docs.px4.io/v1.12/en/ros/ros2_comm.html#build-ros-2-workspace) to get data from the `px4_sitl_rtps gazebo` simulator to cross the micrortps bridge and show up in ROS2.  I could also run the offboard_mode example with ROS2.
Around July 8, the `master` branch of this repo was merged into `main`.  When that occurred, data from the `px4_sitl_rtps gazebo` simulator would be sent over the micrortps bridge, but the micrortps_agent would die with this error:
```
terminate called after throwing an instance of 'eprosima::fastcdr::exception::BadParamException'
  what():  Unexpected byte value in Cdr::deserialize(bool), expected 0 or 1
Aborted (core dumped)
```
I believe the bug is associated with this [issue/fix](https://github.com/PX4/PX4-Autopilot/pull/12025), which was made on the `master` branch, but doesn't appear to be fully integrated into the `main` branch.  The first file seems fine.  The second file doesn't.  I tried editing the second file associated with this [MR](https://github.com/PX4/PX4-Autopilot/pull/12025/files) (which appears to be renamed slightly to `msg/templates/uorb_microcdr/msg.cpp.em`), but was unable to get it to work.

When I revert the repo to an earlier version (i.e., right before the `master` -> `main` merge by using `git checkout 10a2b4c9f7d34234459b4d3604b99a33491a9d83`), everything works properly (i.e., data is sent over the micrortps bridge, the micrortps_agent receives it, and you can `ros2 topic echo` any of the messages).

I have checked whether the `px4_msgs` or `px4_ros_com` repos have changed in the previous month.  They have not.  The source of the miscommunication across the micrortps bridge appears to be associated with recent changes to the PX4-Autopilot repo.

## To Reproduce
Steps to reproduce the behavior:
1. In terminal 1, get the latest version of this repo and start the simulator:
    ```
    git clone https://github.com/PX4/PX4-Autopilot --recursive
    cd ~/PX4-Autopilot
    make
    make px4_sitl_rtps
    make px4_sitl_rtps gazebo
    ```
2. In terminal 2, follow the steps described [here](https://docs.px4.io/v1.12/en/ros/ros2_comm.html#build-ros-2-workspace) to install Fast-DDS, ROS2, and the px4_ros_com_ros2 workspace.  When running the [sanity check the installation](https://docs.px4.io/v1.12/en/ros/ros2_comm.html#sanity-check-the-installation) section, you will run these commands:
    ```
    source ~/px4_ros_com_ros2/install/setup.bash
    micrortps_agent -t UDP
    ```
    which will produce this error:
    ```
    terminate called after throwing an instance of 'eprosima::fastcdr::exception::BadParamException'
      what():  Unexpected byte value in Cdr::deserialize(bool), expected 0 or 1
    Aborted (core dumped)
    ```
3. In terminal 1, revert the PX4-Autopilot to an earlier version and restart the simulator:
    ```
    cd ~/PX4-Autopilot
    git checkout 10a2b4c9f7d34234459b4d3604b99a33491a9d83
    make
    make px4_sitl_rtps
    make px4_sitl_rtps gazebo
    ```
4. In terminal 2, restart the micrortps_agent (`micrortps_agent -t UDP`).  It should not die.
6. In terminal 3, you can now `ros2 topic echo` any of the fields and it will print out data: 
    ```
    ros2 topic echo /fmu/sensor_combined/out
    ---
    timestamp: 1657838030351792
    gyro_rad:
    - -0.0005326324608176947
    - -0.0007989492150954902
    - -0.0007989475270733237
    gyro_integral_dt: 4000
    accelerometer_timestamp_relative: 0
    accelerometer_m_s2:
    - 0.06823474913835526
    - 0.07062912732362747
    - -9.889251708984375
    accelerometer_integral_dt: 4000
    accelerometer_clipping: 0
    gyro_clipping: 0
    accel_calibration_count: 0
    gyro_calibration_count: 1
    ```

## Expected behavior
There should be no error when performing step 2 above.

## Log Files and Screenshots
N/A

## Drone:
px4_sitl_rtps (iris rtps)

## Additional context
None

Noticing that the last commit on [px4_msgs](https://github.com/PX4/px4_msgs) is dated to the 21 of June 2022 while the [history](https://github.com/PX4/PX4-Autopilot/commits/main/msg) of the `msg/` folder of PX4-Autopilot shows that the last modification has been made just 2 days ago (the 20 of July 2022).
For this reason I tried to force the update of the message definitions on the px4_msgs side using the [this](https://github.com/PX4/PX4-Autopilot/blob/main/msg/tools/uorb_to_ros_msgs.py) script. Indeed some message definitions changed.

However at this point I am not able to build px4_msgs, here is the error message:

> CMake Error at /home/beniamino/ros2_foxy/install/rosidl_adapter/share/rosidl_adapter/cmake/rosidl_adapt_interfaces.cmake:60 (message):
>   execute_process(/usr/bin/python3 -m rosidl_adapter --package-name px4_msgs
>   --arguments-file
>   /home/beniamino/px4_ros_com_ros2/build/px4_msgs/rosidl_adapter__arguments__px4_msgs.json
>   --output-dir
>   /home/beniamino/px4_ros_com_ros2/build/px4_msgs/rosidl_adapter/px4_msgs
>   --output-file
>   /home/beniamino/px4_ros_com_ros2/build/px4_msgs/rosidl_adapter/px4_msgs.idls)
>   returned error code 1:
> 
>   Traceback (most recent call last):
> 
>     File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
>       return _run_code(code, main_globals, None,
>     File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
>       exec(code, run_globals)
>     File "/home/beniamino/ros2_foxy/install/rosidl_adapter/lib/python3.8/site-packages/rosidl_adapter/__main__.py", line 19, in <module>
>       sys.exit(main())
>     File "/home/beniamino/ros2_foxy/install/rosidl_adapter/lib/python3.8/site-packages/rosidl_adapter/main.py", line 53, in main
>       abs_idl_file = convert_to_idl(
>     File "/home/beniamino/ros2_foxy/install/rosidl_adapter/lib/python3.8/site-packages/rosidl_adapter/__init__.py", line 19, in convert_to_idl
>       return convert_msg_to_idl(
>     File "/home/beniamino/ros2_foxy/install/rosidl_adapter/lib/python3.8/site-packages/rosidl_adapter/msg/__init__.py", line 28, in convert_msg_to_idl
>       msg = parse_message_string(package_name, input_file.stem, content)
>     File "/home/beniamino/ros2_foxy/install/rosidl_adapter/lib/python3.8/site-packages/rosidl_adapter/parser.py", line 531, in parse_message_string
>       constants.append(Constant(type_string, name, value))
>     File "/home/beniamino/ros2_foxy/install/rosidl_adapter/lib/python3.8/site-packages/rosidl_adapter/parser.py", line 319, in __init__
>       raise NameError(
> 
>   NameError: 'kNotAborted' is an invalid constant name.  It should have the
>   pattern `'^[A-Z]([A-Z0-9_]?[A-Z0-9]+)*$'`


It seems to me that this PR #19871 modified the [position_controller_landing_status](https://github.com/PX4/PX4-Autopilot/blob/6a0f394d463494908c5b4fcbde7b63ab3da66d33/msg/position_controller_landing_status.msg) uorb message and now its constant names are not consistent with the ROS naming conventions (they are camel case while the should be snake case).

If restore just that message definition in px4_msgs than everything works (provided that that message is not exchanged between between micrortps_agent and micrortps_client).

--- a/msg/position_controller_landing_status.msg
+++ b/msg/position_controller_landing_status.msg
@@ -9,8 +9,8 @@ uint8 abort_status
 
 # abort reasons
 # after the manual operator abort, corresponds to individual bits of param FW_LND_ABORT
-uint8 kNotAborted = 0
-uint8 kAbortedByOperator = 1
-uint8 kTerrainNotFound = 2 # FW_LND_ABORT (1 << 0)
-uint8 kTerrainTimeout = 3 # FW_LND_ABORT (1 << 1)
-uint8 kUnknownAbortCriterion = 4
+uint8 NOT_ABORTED = 0
+uint8 ABORTED_BY_OPERATOR = 1
+uint8 TERRAIN_NOT_FOUND = 2 # FW_LND_ABORT (1 << 0)
+uint8 TERRAIN_TIMEOUT = 3 # FW_LND_ABORT (1 << 1)
+uint8 UNKNOWN_ABORT_CRITERION = 4

--- a/src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp
+++ b/src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp
@@ -234,7 +234,7 @@ FixedwingPositionControl::vehicle_command_poll()
 			    _pos_sp_triplet.current.valid &&
 			    (_pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_LAND)) {
 
-				updateLandingAbortStatus(position_controller_landing_status_s::kAbortedByOperator);
+				updateLandingAbortStatus(position_controller_landing_status_s::ABORTED_BY_OPERATOR);
 			}
 
 		} else if (vehicle_command.command == vehicle_command_s::VEHICLE_CMD_DO_CHANGE_SPEED) {
@@ -613,21 +613,21 @@ FixedwingPositionControl::updateLandingAbortStatus(const uint8_t new_abort_statu
 	if (new_abort_status > 0 && _landing_abort_status != new_abort_status) {
 
 		switch (new_abort_status) {
-		case (position_controller_landing_status_s::kAbortedByOperator): {
+		case (position_controller_landing_status_s::ABORTED_BY_OPERATOR): {
 				mavlink_log_critical(&_mavlink_log_pub, "Landing aborted by operator\t");
 				events::send(events::ID("fixedwing_position_control_landing_abort_status_operator_abort"), events::Log::Critical,
 					     "Landing aborted by operator");
 				break;
 			}
 
-		case (position_controller_landing_status_s::kTerrainNotFound): {
+		case (position_controller_landing_status_s::TERRAIN_NOT_FOUND): {
 				mavlink_log_critical(&_mavlink_log_pub, "Landing aborted: terrain estimate not found\t");
 				events::send(events::ID("fixedwing_position_control_landing_abort_status_terrain_not_found"), events::Log::Critical,
 					     "Landing aborted: terrain measurement not found");
 				break;
 			}
 
-		case (position_controller_landing_status_s::kTerrainTimeout): {
+		case (position_controller_landing_status_s::TERRAIN_TIMEOUT): {
 				mavlink_log_critical(&_mavlink_log_pub, "Landing aborted: terrain estimate timed out\t");
 				events::send(events::ID("fixedwing_position_control_landing_abort_status_terrain_timeout"), events::Log::Critical,
 					     "Landing aborted: terrain estimate timed out");
@@ -642,8 +642,8 @@ FixedwingPositionControl::updateLandingAbortStatus(const uint8_t new_abort_statu
 		}
 	}
 
-	_landing_abort_status = (new_abort_status >= position_controller_landing_status_s::kUnknownAbortCriterion) ?
-				position_controller_landing_status_s::kUnknownAbortCriterion : new_abort_status;
+	_landing_abort_status = (new_abort_status >= position_controller_landing_status_s::UNKNOWN_ABORT_CRITERION) ?
+				position_controller_landing_status_s::UNKNOWN_ABORT_CRITERION : new_abort_status;
 	landing_status_publish();
 }
 
@@ -1351,7 +1351,7 @@ FixedwingPositionControl::control_auto_loiter(const float control_interval, cons
 	if (_landing_abort_status) {
 		if (pos_sp_curr.alt - _current_altitude  < _param_fw_clmbout_diff.get()) {
 			// aborted landing complete, normal loiter over landing point
-			updateLandingAbortStatus(position_controller_landing_status_s::kNotAborted);
+			updateLandingAbortStatus(position_controller_landing_status_s::NOT_ABORTED);
 
 		} else {
 			// continue straight until vehicle has sufficient altitude
@@ -2364,7 +2364,7 @@ FixedwingPositionControl::reset_landing_state()
 	// reset abort land, unless loitering after an abort
 	if (_landing_abort_status && (_pos_sp_triplet.current.type != position_setpoint_s::SETPOINT_TYPE_LOITER)) {
 
-		updateLandingAbortStatus(position_controller_landing_status_s::kNotAborted);
+		updateLandingAbortStatus(position_controller_landing_status_s::NOT_ABORTED);
 	}
 }
 
@@ -2668,10 +2668,10 @@ FixedwingPositionControl::getLandingTerrainAltitudeEstimate(const hrt_abstime &n
 
 			const bool terrain_first_measurement_timed_out = (now - _time_started_landing) > TERRAIN_ALT_FIRST_MEASUREMENT_TIMEOUT;
 			const bool abort_on_terrain_measurement_timeout = checkLandingAbortBitMask(_param_fw_lnd_abort.get(),
-					position_controller_landing_status_s::kTerrainNotFound);
+					position_controller_landing_status_s::TERRAIN_NOT_FOUND);
 
 			if (terrain_first_measurement_timed_out && abort_on_terrain_measurement_timeout) {
-				updateLandingAbortStatus(position_controller_landing_status_s::kTerrainNotFound);
+				updateLandingAbortStatus(position_controller_landing_status_s::TERRAIN_NOT_FOUND);
 			}
 
 			return land_point_altitude;
@@ -2681,10 +2681,10 @@ FixedwingPositionControl::getLandingTerrainAltitudeEstimate(const hrt_abstime &n
 
 			const bool terrain_timed_out = (now - _last_time_terrain_alt_was_valid) > TERRAIN_ALT_TIMEOUT;
 			const bool abort_on_terrain_timeout = checkLandingAbortBitMask(_param_fw_lnd_abort.get(),
-							      position_controller_landing_status_s::kTerrainTimeout);
+							      position_controller_landing_status_s::TERRAIN_TIMEOUT);
 
 			if (terrain_timed_out && abort_on_terrain_timeout) {
-				updateLandingAbortStatus(position_controller_landing_status_s::kTerrainTimeout);
+				updateLandingAbortStatus(position_controller_landing_status_s::TERRAIN_TIMEOUT);
 			}
 
 			return _last_valid_terrain_alt_estimate;

--- a/src/modules/fw_pos_control_l1/FixedwingPositionControl.hpp
+++ b/src/modules/fw_pos_control_l1/FixedwingPositionControl.hpp
@@ -338,7 +338,7 @@ private:
 	// [m] relative height above land point
 	float _landing_approach_entrance_rel_alt{0.0f};
 
-	uint8_t _landing_abort_status{position_controller_landing_status_s::kNotAborted};
+	uint8_t _landing_abort_status{position_controller_landing_status_s::NOT_ABORTED};
 
 	bool _flaring{false};
 	hrt_abstime _time_started_flaring{0}; // [us]
@@ -436,7 +436,7 @@ private:
 	 *
 	 * @param new_abort_status Either 0 (not aborted) or the singular bit >0 which triggered the abort
 	 */
-	void updateLandingAbortStatus(const uint8_t new_abort_status = position_controller_landing_status_s::kNotAborted);
+	void updateLandingAbortStatus(const uint8_t new_abort_status = position_controller_landing_status_s::NOT_ABORTED);
 
 	/**
 	 * @brief Checks if the automatic abort bitmask (from FW_LND_ABORT) contains the given abort criterion.

